error: '%s' directive argument is null

From https://github.com/JonathonReinhart/staticx/issues/116#issuecomment-579655182

```
    gcc -o scons_build/release/bootloader/main.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Os -DNDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/main.c
    In function 'create_tmpdir',
        inlined from 'main' at bootloader/main.c:393:20:
    bootloader/main.c:211:9: error: '%s' directive argument is null [-Werror=format-overflow=]
      211 |         error(2, errno, "Failed to create temporary directory: %s", tmpdir);
          |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    cc1: all warnings being treated as errors
```

--- a/bootloader/main.c
+++ b/bootloader/main.c
@@ -205,10 +205,11 @@ patch_app(const char *prog_path)
 static char *
 create_tmpdir(void)
 {
-    char *template = path_join(getenv(TMPDIR) ?: "/tmp", "staticx-XXXXXX");
+    const char *tmproot = getenv(TMPDIR) ?: "/tmp";
+    char *template = path_join(tmproot, "staticx-XXXXXX");
     char *tmpdir = mkdtemp(template);
     if (!tmpdir)
-        error(2, errno, "Failed to create temporary directory: %s", tmpdir);
+        error(2, errno, "Failed to create temporary directory in %s", tmproot);
     assert(tmpdir == template);
     return tmpdir;
 }

